hypothesis-python/src/hypothesis/internal/reflection.py

```diff
@@ -588,3 +595,72 @@ def accept(proxy):
         )
 
     return accept
+
+
+def deprecated_posargs(func: C) -> C:
+    """Insert a `*__deprecated_posargs,` shim in place of the `*,` for kwonly args.
+
+    This turns out to be fairly tricky to get right with our preferred style of
+    error handling (exhaustive) and various function-rewriting wrappers.
+    """
+    if READTHEDOCS:  # pragma: no cover
+        # Documentation should show the new signatures without deprecation helpers.
+        return func
+
+    signature = inspect.signature(func)
+    parameters = list(signature.parameters.values())
+    vararg = inspect.Parameter(
+        name="__deprecated_posargs",
+        kind=inspect.Parameter.VAR_POSITIONAL,
+        annotation=Any,
+    )
+
+    # If we're passed any VAR_POSITIONAL args, we'll use this sequence of
+    # params to match them up with the correct KEYWORD_ONLY argument after
+    # checking that it has it's default value - if you're passing by name,
```

Comment:
```suggestion
    # checking that it has its default value - if you're passing by name,
```

no apostrophe in possessive "its".